src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -8297,6 +8297,40 @@ public void PathMapParser()
             Assert.Equal(1, parsedArgs.Errors.Count());
             Assert.Equal((int)ErrorCode.ERR_InvalidPathMap, parsedArgs.Errors[0].Code);
         }
+
+        [WorkItem(8470, "https://github.com/dotnet/roslyn/issues/8470")]
+        [Fact]
+        public void CompareSymbolDistinguisherDiagnosticArgumentsNoCompilation()
```

Comment:
I think we can reproduce the scenario with "normal" compilation test, by getting the diagnostics and calling get hash code explicitly.